scripts/unreleased-change-monitor.ts

```diff
@@ -0,0 +1,116 @@
+import { Octokit } from 'octokit';
+import { LimaAndQemu, AlpineLimaISO } from 'scripts/dependencies/lima';
+import * as tools from 'scripts/dependencies/tools';
+import { WSLDistro, HostResolverHost } from 'scripts/dependencies/wsl';
+import { Dependency, UnreleasedChangeMonitor, HasUnreleasedChangesResult, getOctokit } from 'scripts/lib/dependencies';
+
+const GITHUB_OWNER = process.env.GITHUB_REPOSITORY?.split('/')[0] || 'rancher-sandbox';
+const GITHUB_REPO = process.env.GITHUB_REPOSITORY?.split('/')[1] || 'rancher-desktop';
+// a (hopefully) unique and communicative key that is used to find issues created by
+// this script by filtering them down to the ones that have it in their title
+const UCMONITOR = 'ucmonitor';
+
+type UnreleasedChangeMonitoringDependency = Dependency & UnreleasedChangeMonitor;
+
+type DependencyState = { dependency: UnreleasedChangeMonitoringDependency } & HasUnreleasedChangesResult;
+
+const dependencies: UnreleasedChangeMonitoringDependency[] = [
+  new LimaAndQemu(),
+  new WSLDistro(),
+  new tools.DockerCLI(),
+  new tools.Steve(),
+  new tools.GuestAgent(),
+  new tools.RancherDashboard(),
+  new AlpineLimaISO(),
+  new HostResolverHost(), // we only need one of HostResolverHost and HostResolverPeer
+];
+
+type Issue = Awaited<ReturnType<Octokit['rest']['search']['issuesAndPullRequests']>>['data']['items'][0];
+
+async function getExistingIssuesFor(dependencyName: string): Promise<Issue[]> {
+  const queryString = `type:issue in:title repo:${ GITHUB_OWNER }/${ GITHUB_REPO } ${ UCMONITOR } ${ dependencyName } sort:updated`;
+  const response = await getOctokit().rest.search.issuesAndPullRequests({ q: queryString });
```

Comment:
Note that this only returns the first page of results; see [documentation](https://www.npmjs.com/package/octokit#pagination) for details.  It would be something like:
```ts
async function* getExistingIssuesFor(dependencyName: string): AsyncIterableIterator<Issue> {
  const q = `type:issue in:title repo:${GITHUB_OWNER}/${GITHUB_REPO} ${UCMONITOR} ${dependencyName} sort:updated`;
  const iterator = getOctokit().paginate.iterator(getOctokit().rest.search.issuesAndPullRequests, { q });
  for await (const { data: issues } of iterator) {
    yield* issues;
  }
}
```
The caller would have to change to `for await (const existingIssue of existingIssues) { …` too, and it's no longer to run them all in parallel (because there can be too many to hold in memory at once).

Reply:
I thought about this too, but I figured I could simplify by using the `sort:updated` clause in the query string. Including that clause will put the most recently updated issues first. Also, this query returns up to 100 issues. All older issues will be closed unless someone manually reopens them, which shouldn't happen. We're really only working with the most recently modified 2-3 issues.

It would be more reliable to go through all the issues on every run, but as far as I can tell that is unnecessary. I'd like to leave it the way it is if you're ok with that.